pkg/dyninst/decode/decoder.go

```diff
@@ -206,35 +206,17 @@ func (d *Decoder) encodeValue(
 			return errors.New("passed data not long enough for pointer")
 		}
 		addr := binary.NativeEndian.Uint64(data)
-		if addr == 0 { // nil pointers are encoded as 0.
-			err := enc.WriteToken(jsontext.String("nil"))
-			if err != nil {
-				return err
-			}
-			return nil
-		}
 		key := typeAndAddr{
 			irType: uint32(v.Pointee.GetID()),
 			addr:   addr,
 		}
 
-		var (
-			header         *output.DataItemHeader
-			pointedData    []byte
-			dontParseValue bool
-		)
+		var address uint64
 		pointedValue, ok := dataItems[key]
```

Comment:
given how far away from here you use this boolean, I'd give it a better name like `found` or `dataFound`?